xrtl/ui/control.h

```diff
@@ -0,0 +1,282 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef XRTL_UI_CONTROL_H_
+#define XRTL_UI_CONTROL_H_
+
+#include <memory>
+#include <mutex>
+#include <utility>
+#include <vector>
+
+#include "xrtl/base/macros.h"
+#include "xrtl/base/ref_ptr.h"
+#include "xrtl/base/threading/message_loop.h"
+#include "xrtl/gfx/color.h"
+
+namespace xrtl {
+namespace ui {
+
+struct Point2D {
```

Comment:
yeah we'll use glm for those though; theses are more specific to UI stuff. I would hope anyone doing heavy work on geometry like this would stick with glm all the way through, whereas the UI stuff is more readable with actual structs instead of for example glm::vec4 as the frame rect.